x/txfees/keeper/hooks_test.go

```diff
@@ -37,49 +37,66 @@ func (suite *KeeperTestSuite) TestTxFeesAfterEpochEnd() {
+		coin      sdk.Coin
+		baseDenom string
+		denom     string
+		poolType  gammtypes.PoolI
+		swapFee   sdk.Dec
+	}{
+		{
+			name:      "TxFees AfterEpochEnd for uion",
+			coin:      sdk.NewInt64Coin(uion, 10),
+			baseDenom: baseDenom,
+			denom:     uion,
+			poolType:  uionPool,
+			swapFee:   sdk.NewDec(0),
+		},
+		{
+			name:      "TxFees AfterEpochEnd for atom",
+			coin:      sdk.NewInt64Coin(atom, 20),
+			baseDenom: baseDenom,
+			denom:     atom,
+			poolType:  atomPool,
+			swapFee:   sdk.NewDec(0),
+		},
+		{
+			name:      "TxFees AfterEpochEnd for ust :(",
+			coin:      sdk.NewInt64Coin(ust, 14),
+			baseDenom: baseDenom,
+			denom:     ust,
+			poolType:  ustPool,
+			swapFee:   sdk.NewDec(0),
+		},
+	}
+
+	var finalOutputAmount = sdk.NewInt(0)
 	moduleAddrFee := suite.App.AccountKeeper.GetModuleAddress(types.FeeCollectorName)
 	moduleAddrNonNativeFee := suite.App.AccountKeeper.GetModuleAddress(types.NonNativeFeeCollectorName)
 
-	// make sure module account is funded with test fee tokens
-	suite.Require().True(suite.App.BankKeeper.HasBalance(suite.Ctx, moduleAddrNonNativeFee, coins[0]))
-	suite.Require().True(suite.App.BankKeeper.HasBalance(suite.Ctx, moduleAddrNonNativeFee, coins[1]))
-	suite.Require().True(suite.App.BankKeeper.HasBalance(suite.Ctx, moduleAddrNonNativeFee, coins[2]))
-
 	params := suite.App.IncentivesKeeper.GetParams(suite.Ctx)
 	futureCtx := suite.Ctx.WithBlockTime(time.Now().Add(time.Minute))
 
-	suite.App.EpochsKeeper.AfterEpochEnd(futureCtx, params.DistrEpochIdentifier, int64(1))
-
-	moduleBaseDenomBalance := suite.App.BankKeeper.GetBalance(suite.Ctx, moduleAddrFee, baseDenom)
-	suite.Require().Empty(suite.App.BankKeeper.GetAllBalances(suite.Ctx, moduleAddrNonNativeFee))
-	suite.Require().True(moduleBaseDenomBalance.Amount.GTE(fullExpectedOutput.Amount))
+	for _, tc := range tests {
+		expectedOutput, err := tc.poolType.CalcOutAmtGivenIn(suite.Ctx,
+			sdk.Coins{sdk.Coin{Denom: tc.denom, Amount: tc.coin.Amount}},
+			tc.baseDenom,
+			tc.swapFee)
+		suite.Require().NoError(err)
+
+		finalOutputAmount = finalOutputAmount.Add(expectedOutput.Amount)
+		_, _, addr0 := testdata.KeyTestPubAddr()
+		simapp.FundAccount(suite.App.BankKeeper, suite.Ctx, addr0, sdk.Coins{tc.coin})
+		suite.App.BankKeeper.SendCoinsFromAccountToModule(suite.Ctx, addr0, types.NonNativeFeeCollectorName, sdk.Coins{tc.coin})
+
+		suite.Require().True(suite.App.BankKeeper.HasBalance(suite.Ctx, moduleAddrNonNativeFee, tc.coin))
+
+		suite.App.EpochsKeeper.AfterEpochEnd(futureCtx, params.DistrEpochIdentifier, int64(1))
+		moduleBaseDenomBalance := suite.App.BankKeeper.GetBalance(suite.Ctx, moduleAddrFee, tc.baseDenom)
+		suite.Require().Empty(suite.App.BankKeeper.GetAllBalances(suite.Ctx, moduleAddrNonNativeFee))
+		// Tried to make use Equal() to check the exact same value but moduleBaseDenomBalance.Amount seems to be incrementing
+		// by 2000000 per test hmm
+		suite.Require().True(moduleBaseDenomBalance.Amount.GTE(finalOutputAmount))
```

Comment:
Tried to make use Equal() to check the exact same value but moduleBaseDenomBalance.Amount seems to be incrementing by 2000000 per test 🤔

Reply:
it's probably because the variable `finalOutputAmount` is defined outside the test cases iteration. Try moving the variable within the test loops! OR this assertion can be moved outside the loops